lms/djangoapps/student_account/views.py

```diff
@@ -380,6 +380,26 @@ def _third_party_auth_context(request, redirect_to, tpa_hint=None):
                 context['errorMessage'] = _(unicode(msg))  # pylint: disable=translation-of-non-string
                 break
 
+        if enterprise_customer and context['errorMessage']:
```

Comment:
It seems odd that we are making this code enterprise-specific. Is there a way for us to refactor this so that we are getting the administrator descriptor for the error message from some separate function that would take the EnterpriseCustomer into account if one is available in the current context and if not maybe use the name of the IdP?

Reply:
@douglashall unfortunately this code is already making use of enterprise_customer I am just use the existing 
object for `enterprise_customer` to update the error message for enterprises.

For refactoring I can move this logic to a method which can take error message and enterprise customer like `_get_error_message_for_enterprise(error_message, enterprise_customer=None)` and will return the updated message.

I believe the method [`enterprise_customer_for_request(request)`](https://github.com/edx/edx-platform/blob/master/openedx/features/enterprise_support/api.py#L375-L395) used above to fetch enterprise customer automatically take care of getting enterprise from `tpa_hint` or `pipeline` or `cookies`.